packages/core/useCycleList/index.ts

```diff
@@ -27,11 +28,13 @@ export interface UseCycleListOptions<T> {
 export function useCycleList<T>(list: T[], options?: UseCycleListOptions<T>): UseCycleListReturn<T> {
   const state = shallowRef(options?.initialValue ?? list[0]) as Ref<T>
 
+  const stateIndex = ref(list.indexOf(state.value))
```

Comment:
this no effect on current problem.only reacting indexOf(state.value).
best work is focus on self index instead of state